src/app/core/entity/entity-config.service.spec.ts

```diff
@@ -134,6 +122,49 @@ describe("EntityConfigService", () => {
     expect(Test.icon).toBe("users");
     expect(Test.color).toBe("red");
   });
+
+  it("should create a new subclass with the schema of the extended", () => {
+    const schema = { dataType: "string", label: "Dynamic Property" };
+    mockConfigService.getAllConfigs.and.returnValue([
+      {
+        _id: "entity:DynamicTest",
+        label: "DynamicTest",
+        extends: "Test",
+        attributes: [{ name: "dynamicProperty", schema }],
+      },
+    ]);
+
+    service.setupEntitiesFromConfig();
+
+    const dynamicEntity = entityRegistry.get("DynamicTest");
+    expect(dynamicEntity.ENTITY_TYPE).toBe("DynamicTest");
+    expect([...dynamicEntity.schema.entries()]).toEqual(
+      jasmine.arrayContaining([...Test.schema.entries()])
+    );
+    expect(dynamicEntity.schema.get("dynamicProperty")).toBe(schema);
+    const dynamicInstance = new dynamicEntity("someId");
+    expect(dynamicInstance instanceof Test).toBeTrue();
```

Comment:
but Test should still be usable without the extension in parallel. Does this work if the extended is still also instanceof Test?
Let's add an expectation that Test does not have the dynamicProperty here to be safe?